### PR TITLE
wpcom-block-editor: Fix the category name is missing when inserting a pattern

### PR DESCRIPTION
**Pull request by arthur791004**
Related to https://github.com/Automattic/dotcom-forge/issues/2176

## Proposed Changes

* Update the query selector so we're able to get the correct category name.

| wpcom_pattern_inserted | wpcom_block_inserted |
| - | - |
| image | image |

## Testing Instructions

* cd `apps/wpcom-block-editor` and run `yarn dev --sync` to apply the changes to your sandbox
* Sandbox widgets.wp.com
* Go to the site editor
* Insert a pattern
* Verify the event is fired with the correct `category_name`. Note that if you insert a pattern from the search bar, the `category_name` will be `null` since you never select a category.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: apps/wpcom-block-editor/src/wpcom/features/tracking.js [hunk 1]
```diff
@@ -272,19 +282,29 @@ const maybeTrackPatternInsertion = ( actionData, additionalData ) => {
...
+		const patternCategory = patternCategories.find(
+			( { label } ) => label === categoryElement?.ariaLabel
+		);
 
 		tracksRecordEvent( 'wpcom_pattern_inserted', {
 			pattern_name: patternName,
-			pattern_category: patternCategory,
+			pattern_category: patternCategory?.name,
 			blocks_replaced,
+			insert_method,
```
Comment (okmttdhr) [inline apps/wpcom-block-editor/src/wpcom/features/tracking.js:297]: 👍

**Review by okmttdhr — APPROVED** [review]
Tested and worked well 👍

**Comment by autumnfjeld** [comment]
Going to have a look to see why null is  being recorded here:
https://mc.a8c.com/tracks/trends/?compareby=prop&eventname=wpcom_pattern_inserted&interval=day&prop=pattern_category&startdate=20230329&stat=uniques
image

**Comment by arthur791004** [comment]
If the user inserts a pattern from the search bar, there isn't any selected category so the null will be recorded.

| Pattern Inserter | Pattern Explorer | The `+` button |
| - | - | - | 
| image | image | image |

**Pull request merged**